src/inmanta/agent/agent.py

```diff
@@ -236,17 +408,18 @@ async def execute(
                     self.running = False
 
             async with self.scheduler.ratelimiter:
-                ctx = handler.HandlerContext(self.resource, logger=self.logger)
+
+                ctx = handler.HandlerLogger(self.resource_id, logger=self.logger)
 
                 ctx.debug(
                     "Start run for resource %(resource)s because %(reason)s",
-                    resource=str(self.resource.id),
+                    resource=str(self.resource_id),
                     deploy_id=self.gid,
                     agent=self.scheduler.agent.name,
                     reason=self.reason,
                 )
```

Comment:
Just had a quick chat with Wouter and we believe this can be safely moved to the executor as well. If I understand correctly this would then completely remove the need for the `HandlerLogger` hack, so I think we can revert the changes we made there.